core/app/models/spree/store_credit_event.rb

```diff
@@ -12,9 +12,9 @@ class StoreCreditEvent < Spree::Base
 
     belongs_to :store_credit
     belongs_to :originator, polymorphic: true
-    belongs_to :update_reason, class_name: "Spree::StoreCreditUpdateReason"
+    belongs_to :store_credit_reason, class_name: "Spree::StoreCreditReason"
 
-    validates_presence_of :update_reason, if: :action_requires_reason?
+    validates_presence_of :store_credit_reason_id, if: :action_requires_reason?
```

Comment:
I think we can use `store_credit_reason` here

Reply:
done